### ISSUE TITLE
Search feature

### ISSUE THREAD

**Issue by xmclark** [tranhungt/okcupidjs#11]
Searching for users similar to how one can browse users on the `/match` route would be useful. A payload with search filters could be used to prune results.

**Comment by tranhungt** [comment]
This would definitely be very useful.

**Comment by xmclark** [comment]
hello @tranhungt. I think I need a little help with git branch workflow (still new for me). I want to push this feature from a remote branch off of my fork (it's called searchFeature I believe). So far I have been pushing from master, but I want to stop doing that for larger changes.

I'm not too familiar with the branching workflow in Git. I understand that I can make a pull request from my feature branch to your master, but how do my changes get integrated back into my fork's master, and then my local master?

Feel free to point me to a resource or video of that's easier. And thank you for patience!!

**Comment by tranhungt** [comment]
Hey @xmclark 
From what you're saying, I'm going to assume that you have your own fork of this project, which I do see under your profile.
That means there exists two different remote sources, my tranhungt/okcupidjs and your xmclark/okcupidjs fork, `remote` being the repositories on github.com.

So to start develop, what you need to ensure is that your fork is up to date compared to mine.
https://help.github.com/articles/syncing-a-fork/

Also, in your terminal, type in `git remote -v`, and it will show you what remotes you are tracking.
The usual naming convention is that `tranhungt/okcupidjs` is named `upstream` and your personal fork `xmclark/okcupidjs` is named `origin`.

From then, check to make sure you're in master branch, by typing `git status`, it will tell you what branch you're on. If you're somehow now on master, then you can switch to it by using `git branch master`. And then ensure that your local master branch is up to date with `git pull origin master`, which will pull from your remote fork, which by now we assume is also up to date with my remote.

Now you can create your NEW feature branch with `git branch -b searchFeature`.
Develop as you would.
When you are done committing, you can push it to your remote fork with `git push origin searchFeature`, and it will automatically create the branch `searchFeature` in your remote.

When you go onto github, and into your okcupidjs project, you can select the `searchFeature` branch, and create a "pull request" as you have done before into my `tranhungt/okcupidjs` repository.
At that point, once I merge your pull request, and you then again sync your master branch from your fork with mine.

Let me know if that makes sense..

**Comment by xmclark** [comment]
Hey @tranhungt thanks for this detailed explanation. I was about 70% of the way there, and you filled in the last 30% for me. 

I followed the links and was able to add the upstream remote. This was the big piece I was missing, because up until now, I've only been looking at my fork and my local. I was able to do a merge into my local master, and I got all the changes. 

I also noticed that I wasn't pushing back up to my fork, so I did a `git push` and everything looks good now. I now have `master`, `upstream/master`, `origin/master` all pointing to the same commit, so I'm in good shape now.

I pushed my searchFeature branch up to origin and just made PR #14.

Thanks for you help!!
